app/src/main/java/com/realityexpander/tasky/agenda_feature/data/repositories/agendaRepository/agendaRepositoryImpls/AgendaRepostoryImpl.kt

```diff
@@ -0,0 +1,67 @@
+package com.realityexpander.tasky.agenda_feature.data.repositories.agendaRepository.agendaRepositoryImpls
+
+import com.realityexpander.tasky.agenda_feature.data.common.convertersDTOEntityDomain.toDTO
+import com.realityexpander.tasky.agenda_feature.data.repositories.agendaRepository.remote.IAgendaApi
+import com.realityexpander.tasky.agenda_feature.data.repositories.eventRepository.eventRepositoryImpls.EventRepositoryImpl
+import com.realityexpander.tasky.agenda_feature.domain.AgendaItem
+import com.realityexpander.tasky.agenda_feature.domain.AgendaSync
+import com.realityexpander.tasky.agenda_feature.domain.IAgendaRepository
+import com.realityexpander.tasky.agenda_feature.domain.IEventRepository
+import com.realityexpander.tasky.agenda_feature.util.EventId
+import java.time.ZonedDateTime
+
+class AgendaRepositoryImpl(
+    private val eventRepository: IEventRepository = EventRepositoryImpl(),
+//    private val taskRepository: ITaskRepository,                                  // todo implement tasks repo
+//    private val reminderRepository: IReminderRepository,                          // todo implement reminders repo
+    private val agendaApi: IAgendaApi,
+) : IAgendaRepository {
+
+    override suspend fun getAgendaDay(dateTime: ZonedDateTime): List<AgendaItem> {
+        val events = eventRepository.getEventsForDay(dateTime)
+//        val tasks = taskRepository.getTasks(dateTime)                             // todo implement tasks repo
+//        val reminders = reminderRepository.getReminders(dateTime)                 // todo implement reminders repo
+        return events // + tasks + reminders
+    }
+
+    override suspend fun syncAgenda(): Boolean {
+        val deletedEventIds = eventRepository.getDeletedEventIds()
+//        val deletedTaskIds = taskRepository.getDeletedTaskIds()                   // todo implement tasks repo
+//        val deletedReminderIds = reminderRepository.getDeletedReminderIds()       // todo implement reminders repo
+
+        val deletedSuccessfully =
+            agendaApi.syncAgenda(
+                AgendaSync(
+                    deleteEventIds = deletedEventIds,
+//                    deleteTaskIds = deletedTaskIds,                               // todo implement tasks repo
+//                    deleteReminderIds = deletedReminderIds,                       // todo implement reminders repo
+                ).toDTO()
+            )
+
+        if (deletedSuccessfully) {
+           return eventRepository.deleteFinallyEventIds(deletedEventIds)
+        }
+
+        return false
+    }
```

Comment:
For this function it's very unclear to me what the return boolean represents. Could that maybe be reflected by either renaming the function or the return value (in a sealed interface maybe)?

Reply:
OK thats a good idea, i will make this change.